src/agent.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""The Jenkins agent relation observer."""
+import logging
+import typing
+
+from ops.charm import CharmBase, RelationJoinedEvent
+from ops.framework import Object
+from ops.model import ActiveStatus, BlockedStatus, Container, MaintenanceStatus
+
+import jenkins
+from state import State
+
+logger = logging.getLogger(__name__)
+
+
+class AgentRelationData(typing.TypedDict):
+    """Relation data required for adding the Jenkins agent.
+
+    Attrs:
+        url: The Jenkins server url.
+        secret: The secret for agent node.
+    """
+
+    url: str
+    secret: str
+
+
+class Observer(Object):
```

Comment:
I've received a feedback that importing as `agent.Observer` helps with clarity, since we can directly know where the import is coming from. Also, since we're importing from the module, it would be better not to repeat the name `Agent`. Would this be okay?

Reply:
That's ok as we don't have any explicit convention for this. Indeed using `agent.Observer` makes it clear, I just find it difficult when you're in a module to understand the context of an object only from the module name, at least for class names.